pkg/endpoints/pipeline.go

```diff
@@ -403,8 +555,42 @@ func getDateTimeAsString() string {
 	return strconv.FormatInt(time.Now().Unix(), 10)
 }
 
+// defines and creates a resource of a specifed type and returns a pipeline resource reference for this resource
+// takes a manual pipeline run data struct, namespace for the resource creation, resource name to refer to it and the resource type
```

Comment:
It seems weird that this `createPipelineResource` function requires a `ManualPipelineRun` parameter. It is possible to create Tekton PipelineResources by itself. So, I think it would be clearer to rename the function to indicate that its use is intended for creating PipelineResources for a given `ManualPipelineRun`. For example, maybe a name like `createPipelineResourceForPipelineRun` would be more clear?